auth/src/main/java/com/firebase/ui/auth/IdpResponse.java

```diff
@@ -124,7 +150,19 @@ public String getIdpSecret() {
      * Get the error code for a failed sign in
      */
     public int getErrorCode() {
-        return mErrorCode;
+        if (isSuccessful()) {
+            return Activity.RESULT_OK;
```

Comment:
Was it possible for this method to return `RESULT_OK` before?  Seems poorly named as `getErrorCode` since it's really `getResultCode`.

However if this was possible before it's not worth breaking the API to fix it.

Reply:
Nope, I did this to keep things backwards compatible. See https://github.com/firebase/FirebaseUI-Android/pull/1010#discussion_r151534213, I'd rather we not use error codes anyway.

Here's how we did it before (now it's clear how illogical that was 😆):
![image](https://user-images.githubusercontent.com/9490724/32915803-f7c8e0ae-cace-11e7-9bbb-3282cd3386d8.png)